pyproject.toml

```diff
@@ -0,0 +1,25 @@
+[tool.poetry]
+name = "vtp-rcv-demo"
+version = "0.1.0"
+description = "Web server to demonstrate the use of VTP+ for RCV elections"
+authors = ["Neil Johnson <neil@cadent.com>"]
+license = "GNU General Public License v2.0"
+readme = "README.md"
+packages = [{include = "vtp_rcv_demo"}]
```

Comment:
Change this?

Reply:
I believe we want to go with **vtp-spring-demo**?

Reply:
yes.  sure.

Reply:
It turns out that I needed a name for the project that doesn't have hyphens, so I used the more descriptive "vtpserver" as the project name. This name works well for imports, and describes the function of the code much more accurately.

We can call the repo whatever you like, and use hyphens, too. So "vtp-spring-demo" or "vtp-sprint-demo" both work, but I'd actually argue for "vtp-api-server" because that's really what this is going to be.

Reply:
That is fine - I lean to more words such as vtpwebserver so to specify what type of server, but either works.

Reply:
Given that this a project called "sprint-demo" I would call it "demo-server". I wouldn't call it just "server" (there's all kinds of servers), or web-server (web servers are so common as to be the default). If it ultimately is intended to be a general VTP API then I'd call it "api-server" (though if that's the case it should end up in a different repository post-demo).

Other points to consider:

- Be aware that the PyPI rules for package naming are deliberately loose. `vtp-project`, `vtp.project`, and `vtp_project` are all the same as far as PyPI is concerned and `pip install` will work with any of them. Most people use dashes. There's also a case to be made for dots in the sense that it implies these are all part of a bigger project (`vtp.api-server`, an `vtp.test-suite` would obviously be in the same project, which is less clear if it's all dashes). I avoid underscores *because* dashes are not allowed in Python identifiers and there's no chance of confusion if they are used.
    - If I am wearing my (**informal!**) BDFL-on-naming hat, I'd call everything `vtp.{project-name-with-dashes}`.

- Please use *some* separator whatever you choose. wordsclusteredtogetherarehardtoread.

- The name can be changed up until the point it goes up on PyPI. I would strongly recommend *not* putting on PyPI until you are willing to commit to a naming scheme for projects as a whole. They cannot be changed easily.

- Last but not least: I recommend a read through [PEP 423 - Naming conventions and recipes related to packaging](https://peps.python.org/pep-0423/), specifically [Top-level namespace relates to code ownership](https://peps.python.org/pep-0423/#top-level-namespace-relates-to-code-ownership).

Reply:
Here's what I found:

- [PEP 8 Package and Module Names](https://peps.python.org/pep-0008/#package-and-module-names] guidelines are succinct and quoted below. Recommended: underscores only in module names, if needed, but not in package names.
- Searching [PyPI](https://pypi.org/), I only found one package that includes the string "vtp": [vtp-elastic-apm 5.1.1](https://pypi.org/project/vtp-elastic-apm/). Note that "vtp" is not part of the namespace hierarchy, and after a brief perusal of this project's docs, I still can't tell what VTP stands for, either.

## Conclusion: "vtp" is a good top-level namespace for VoteTrackerPlus

The string "vtp" doesn't appear to be in use very much at all on PyPI, and is certainly not being used as a top-level namespace. Therefore, we could proceed ahead with using "vtp" as the top-level of the VoteTrackerPlus namespace, even if we never add any of the related repos to PyPI.

For this specific package, I favor the namespace: `vtp.api` as the shortest possible option, or `vtp.api_server` as a more descriptive option.

The repo name can be whatever we want, but I'm leaning toward something more functionally descriptive, like "VTP-API", or "VTP-API-Demo" if we want to manage expectations.

Here's the full text of the [PEP 8 Package and Module Names](https://peps.python.org/pep-0008/#package-and-module-names] guidelines:

> Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged.

Reply:
Based on today's meeting, we've articulated a namespace that works for VTP and applied it to this repo.